user_sync/connector/directory_ldap.py

```diff
@@ -393,6 +393,12 @@ def get_member_groups(self, user):
         """
         group_names = []
         groups = LDAPValueFormatter.get_attribute_value(user, 'memberOf')
+
+        if not groups:
+            return group_names
+        elif type(groups) is str:
```

Comment:
We should generally use `instanceof` to do type checking instead of `type` since `instanceof` will account for inheritance.

Example:

```python
if instanceof(groups, str):
  groups = [groups]
```

(actually, duck-type checking is the most preferred way of doing it, but it might be difficult to distinguish between a string and array without checking the type)

Reply:
Done!